lib/collectors.js

```diff
@@ -0,0 +1,35 @@
+function collectHeaders (context, queryId, response) {
+  context.collectors.responseHeaders ??= {}
+  context.collectors.responseHeaders[queryId] = response.headers
+}
+
+function collectStatusCode (context, queryId, response) {
```

Comment:
for consistency

```suggestion
function collectStatusCodes (context, queryId, response) {
```

Reply:
actually maybe not, since a single service will have a single status code. feel free to ignore this!